filesystem.go

```diff
@@ -353,3 +311,45 @@ func (f FilesystemObjectiveServer) GetREDRequests(ctx context.Context, expr, gro
 func (f FilesystemObjectiveServer) GetREDErrors(ctx context.Context, expr, grouping string, i, i2 int32) (openapiserver.ImplResponse, error) {
 	return openapiserver.ImplResponse{}, errEndpointNotImplemented
 }
+
+// objectiveAsRuleFile reads a ServiceLevelObjective YAML manifest and outputs the corresponding
+// Prometheus rules as a file in the desired directory.
+func objectiveAsRuleFile(file, prometheusFolder string) (slo.Objective, error) {
```

Comment:
It's probably cleaner if we can somehow figure out a way to only do the writes in this function and return the error at the end. The `slo.Objective` could be created outside this function? 
Probably we'd have a read and write (many) function?

Reply:
I tried splitting it into two separate functions, but not we're reading SLO resources twice 😅 

Is that what you had in mind?